### PR TITLE
Add test for not required e164 (phonenumber)

### PR DESCRIPTION
**Pull request by genofire**
## Fixes
or better create an Test to make it visible an fix is necessary

**Make sure that you've checked the boxes below before you submit PR:**
- [x] Tests exist or have been written that cover this particular change.

@go-playground/validator-maintainers

### CODE REVIEW

**Comment by deankarn** [comment]
I'm not sure what this PR is attempting to show. Is `omitempty` not what you are looking for?

**Comment by genofire** [comment]
Do you mean `json:"omitempty"` ? This just has impact of Marshalling ... It is still jnternal an empty string.

It just show, that the current regex solution also run on empty values. On this way it is an indirect `required` and a useless library for optional values.

**Comment by deankarn** [comment]
@genofire  please read the documentation, omitempty is what your are missing if you want empty or a valid value

File: validator_test.go [hunk 1]
```diff
@@ -5515,7 +5515,7 @@ func TestStructOnlyValidation(t *testing.T) {
 		LastName          string     `json:"lname"`
 		Age               uint8      `validate:"gte=0,lte=130"`
 		Number            string     `validate:"required,e164"`
-		AlternativeNumber string     `validate:"e164"`
+		AlternativeNumber string     `validate:"e164,omitempty"`
```
Comment (deankarn) [inline validator_test.go:5517]: This is NOT how omit empty work with validator. PLEASE read the documentation provided.

`omitempty` comes before whatever you want that to be true for.

Comment (genofire) [reply validator_test.go:5517]: Where is it dritten, that `omitempty` needs to be before other validators?

```
Allows conditional validation, for example if a field is not set with a value (Determined by the "required" validator) then other validation such as min or max won't run, but if a value is set validation will run.
```

**Comment by genofire** [comment]
Why is it closed?

**Pull request closed without merging**